src/main/scala/xiangshan/cache/mmu/BitmapCheck.scala

```diff
@@ -0,0 +1,431 @@
+  val enq_ptr_reg = RegNext(enq_ptr)
+
+  val need_addr_check = RegNext(enq_state === state_addr_check && io.req.fire && !flush)
+
+  io.pmp.req.valid := need_addr_check 
+  io.pmp.req.bits.addr := RegEnable(getBitmapAddr(io.req.bits.bmppn),io.req.fire)
+  io.pmp.req.bits.cmd := TlbCmd.read
+  io.pmp.req.bits.size := 3.U 
+  val pmp_resp_valid = io.pmp.req.valid 
+
+  when (io.req.fire) {
+    state(enq_ptr) := enq_state
+    entries(enq_ptr).ppn := io.req.bits.bmppn
+    entries(enq_ptr).vpn := io.req.bits.vpn
+    entries(enq_ptr).id := io.req.bits.id
+    entries(enq_ptr).wait_id := Mux(to_wait, wait_id, enq_ptr)
+    entries(enq_ptr).cf := false.B
+    for (i <- 0 until tlbcontiguous) {
+      entries(enq_ptr).cfs(i) := false.B
+    }
+    entries(enq_ptr).hit := to_wait
+    entries(enq_ptr).level := io.req.bits.level
+    entries(enq_ptr).way_info := io.req.bits.way_info
+    entries(enq_ptr).hptw_bypassed := io.req.bits.hptw_bypassed
+  }
+
+  // when pmp check failed,use cf bit represent
+  when (pmp_resp_valid) {
+    val ptr = enq_ptr_reg
+    val accessFault = io.pmp.resp.ld || io.pmp.resp.mmio
+    entries(ptr).cf := accessFault
+    for (i <- 0 until tlbcontiguous) {
+      entries(ptr).cfs(i) := accessFault
+    }
+    // firstly req bitmap cache
+    state(ptr) := Mux(accessFault, state_mem_out, state_cache_req)
+  }
+
+  val cache_wait = ParallelOR(is_cache_resp).asBool
+  io.cache.resp.ready := !flush && cache_wait
+
+  val hit = WireInit(false.B)
+  io.cache.req.valid := cache_req_arb.io.out.valid && !flush
+  io.cache.req.bits.tag := cache_req_arb.io.out.bits.tag
+  io.cache.req.bits.order := cache_req_arb.io.out.bits.order
+  cache_req_arb.io.out.ready := io.cache.req.ready
+  
+
+  when (cache_req_arb.io.out.fire) {
+    for (i <- state.indices) {
+      when (state(i) === state_cache_req && cache_req_arb.io.chosen === i.U) {
+        state(i) := state_cache_resp
+      }
+    }
+  }
+
+  when (io.cache.resp.fire) {
+    for (i <- state.indices) {
+      val cm_dup_vec = state.indices.map(j =>
+        dupBitmapPPN(entries(i).ppn, entries(j).ppn)
+      )
+      val cm_dup_req_fire = mem_arb.io.out.fire && dupBitmapPPN(entries(i).ppn, mem_arb.io.out.bits.ppn)
+      val cm_dup_vec_wait = cm_dup_vec.zip(is_waiting).map{case (d, w) => d && w}
+      val cm_dup_wait_resp = io.mem.resp.fire && VecInit(cm_dup_vec_wait)(io.mem.resp.bits.id - (l2tlbParams.llptwsize + 2).U)
+      val cm_wait_id = Mux(cm_dup_req_fire, mem_arb.io.chosen, ParallelMux(cm_dup_vec_wait zip entries.map(_.wait_id)))
+      val cm_to_wait = Cat(cm_dup_vec_wait).orR || cm_dup_req_fire
+      val cm_to_mem_out = cm_dup_wait_resp
+      val cm_next_state_normal = MuxCase(state_mem_req, Seq(
+        cm_to_mem_out -> state_mem_out,
+        cm_to_wait -> state_mem_waiting
+      ))
+      when (state(i) === state_cache_resp && io.cache.resp.bits.order === i.U) {
+          hit := io.cache.resp.bits.hit
+          when (hit) {
```

Comment:
`when(???) {`

Reply:
Here it is OK for MMU because of MMU's past formatting style. In the future, the formatting of MMU and other modules will be harmonised.